### PR TITLE
Add GitHub pages

### PR DESCRIPTION
**Pull request by remcohaszing**
This requires additional changes in the repository GitHub Pages settings. _Source_ needs to be set to _GitHub Actions_, and a custom domain needs to be configured.

### CODE REVIEW

File: .github/workflows/website.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: website
+on:
+  push:
+    branches: [main]
```
Comment (wooorm) [inline .github/workflows/website.yml:4]:
```suggestion
    branches:
      - main
```

File: .github/workflows/website.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+    environment:
+      name: Website
+      url: ${{ steps.deployment.outputs.page_url }}
```
Comment (wooorm) [inline .github/workflows/website.yml:14]:
```suggestion
      url: ${{steps.deployment.outputs.page_url}}
```

File: .github/workflows/website.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with: {node-version: 18}
```
Comment (wooorm) [inline .github/workflows/website.yml:18]:
```suggestion
        with:
          node-version: 18
```

Comment (wooorm) [reply .github/workflows/website.yml:18]: Why 18? `node` is the always stable one. Perhaps that?

**Review by wooorm — COMMENTED** [review]
One Q rest stylistic

File: .github/workflows/website.yml (hunk 3 above)
Comment (remcohaszing) [reply .github/workflows/website.yml]: I copy-pasted this and forgot to update.

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
>  a custom domain needs to be configured.

Do you mean a `CNAME` file in `public/`?

**Comment by remcohaszing** [comment]
No, when using GitHub actions as a source, the custom domain must be configured under the pages settings.

**Comment by wooorm** [comment]
I think that’s the same? when you change the custom domain in the interface, it writes to a CNAME file? Or at least it always used to be that way?

**Comment by remcohaszing** [comment]
I found it in the [docs](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain) now:

> Under "Custom domain", type your custom domain, then click **Save**. If you are publishing your site from a branch, this will create a commit that adds a `CNAME` file directly to the root of your source branch. If you are publishing your site with a custom GitHub Actions workflow, no `CNAME` file is created. For more information about your publishing source, see "[Configuring a publishing source for your GitHub Pages site](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)."

**Pull request merged**